### PR TITLE
Many many relationship

### PR DESCRIPTION
**Pull request by JOdiemo**
## Learning objectives
- Understand the different types of relationships between tables.
- Prepare complex queries that answer analytical questions.
- Use primary key & foreign key mechanism for joining tables.

### The following actions is what entails this task

- Create a table named `vets` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
  - `age`: integer
  - `date_of_graduation`: date
- There is a **many-to-many** relationship between the tables `species` and `vets`: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called `specializations` to handle this relationship.
- There is a **many-to-many** relationship between the tables `animals` and `vets`: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called `visits` to handle this relationship, it should also keep track of the date of the visit.
- Insert the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- Insert the following data for specializations:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.
- Insert the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.
 
##Screenshots to the above are in the below link

https://docs.google.com/document/d/1a3CoM6GNrqbfs8_mYQmqBW-9PddcAKI4/edit?usp=sharing&ouid=118251809010617147508&rtpof=true&sd=true

- Write queries to answer the following:
  - Who was the last animal seen by William Tatcher?
  - How many different animals did Stephanie Mendez see?
  - List all vets and their specialties, including vets with no specialties.
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  - What animal has the most visits to vets?
  - Who was Maisy Smith's first visit?
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many visits were with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
  - 
##Screenshot of the results of the questions.**
https://docs.google.com/document/d/1KIpEywXxEKETK3F6Bt_uKric4x2Kmzef/edit?usp=sharing&ouid=118251809010617147508&rtpof=true&sd=true

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -51,3 +51,35 @@ ALTER TABLE animals
 	ON DELETE CASCADE;	
 
 select * from animals
+
+/*Day Four*/
```
Comment (TadesseDev) [inline schema.sql]: - [ ] Dear @JOdiemo, Please move these and all other select statements to the `queries.sql` file. Remember, everything that is a query that helps to analyze data should be saved in the `queries.sql` file.

File: queries.sql [hunk 2]
```diff
@@ -100,3 +100,59 @@ SELECT * FROM animals INNER JOIN owners ON animals.owner_id = owners.id where ow
...
+
+/*Day Four*/
+
+/*Questions*/
+
+--Who was the last animal seen by William Tatcher?
+select animals.name, vets.name, visits.date_of_visit FROM animals 
+	INNER JOIN visits ON animals.id = visits.animals_id
+	LEFT JOIN vets ON vets.id = visits.vet_id 
+	WHERE vets.id = 5 ORDER BY visits.date_of_visit DESC LIMIT 1
```
Comment (TadesseDev) [inline queries.sql]: - [Optional] Dear @JOdiemo, Here we are looking for  the last animal seen by `William Tatcher`, hence we should use the name to look for the record. Something like `WHERE vets.name = 'William Tatcher'`

File: queries.sql [hunk 3]
```diff
@@ -100,3 +100,59 @@ SELECT * FROM animals INNER JOIN owners ON animals.owner_id = owners.id where ow
...
+	WHERE vets.id = 7 GROUP BY vets.name
+--List all vets and their specialties, including vets with no specialties.
+SELECT vets.name, species.name FROM vets
+	LEFT JOIN specializations ON vets.id = specializations.vet_id
+	LEFT JOIN species ON specializations.species_id = species.id  
+--List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT animals.name, vets.name, visits.date_of_visit FROM animals
+	inner JOIN visits ON animals.id = visits.vet_id
+	RIGHT JOIN vets ON vets.id = visits.vet_id 
+	WHERE vets.id= 7 AND
```
Comment (TadesseDev) [inline queries.sql]: - [Optional] Here also, My previous comment about using the name instead of the id is applicable.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @JOdiemo 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @JOdiemo 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ `schema.sql` is updated
-  ✅ Great work using the right GitFlow 
-  ✅ Your README looks good, well done! 

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**